mirgecom/wall_model.py

```diff
@@ -0,0 +1,263 @@
+""":mod:`mirgecom.multiphysics.wall_model` handles the EOS for wall model.
+
+.. autoclass:: PorousFlowDependentVars
+.. autoclass:: WallEOS
+.. autoclass:: WallDependentVars
+.. autoclass:: WallDegradationModel
+"""
+
+from dataclasses import dataclass
+from abc import abstractmethod
+from typing import Union
+import numpy as np
+from meshmode.dof_array import DOFArray
+from arraycontext import dataclass_array_container
+from mirgecom.fluid import ConservedVars
+from mirgecom.eos import MixtureDependentVars
+from mirgecom.transport import GasTransportVars
+
+
+@dataclass_array_container
+@dataclass(frozen=True)
+class PorousFlowDependentVars(MixtureDependentVars):
+    """Dependent variables for the wall model.
+
+    .. attribute:: wall_density
+    """
+
+    wall_density: Union[DOFArray, np.ndarray]
```

Comment:
Why `np.ndarray`?

Reply:
It can be one material (carbon fiber, oxidation) or multiple components (resin + fiber, phenolics)

Reply:
Would it make sense to call it something like `wall_densities` or `wall_component_density` then, to make that more obvious? (Or should it be `*_mass` instead of `*_density`? I'm forgetting how our naming convention applies here @MTCam.)

Reply:
For the base/raw CV array container, we name the components:
`mass`, `momentum`, `energy`, and `species mass` after the names of the conservation equations.  The CV array container is appropriate for storing conservation-equation-associated variables whether they are scalar, vector, or tensor.

For the FluidState, which specifically has a CV object that contains the conserved fluid state, and a DV object with the state-dependent fluid quantities, we name the components after what they actually are:
`mass_density`, `momentum_density`, `energy_density`, and `species_mass_density`

Reply:
Renamed to `materials_density` in [b9d8385](https://github.com/illinois-ceesd/mirgecom/pull/875/commits/b9d8385d0228402dad0cb5caf387cf5ff3cb0086)